### PR TITLE
FS:4651:  Add Form to import new templates

### PR DESCRIPTION
**Pull request by NarenderRajuB**
FS:4651:  Add Form to import new templates

### Change description
_A brief description of the pull request_

- [ x] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
1. Click of View templates  will show below screen, Enter template name and select json file  and click to add the template.

image
image
image
image

### CODE REVIEW

File: app/blueprints/templates/routes.py [hunk 1]
```diff
@@ -12,9 +18,32 @@
...
     sections = get_all_template_sections()
     forms = get_all_template_forms()
-    return render_template("view_templates.html", sections=sections, forms=forms)
+    form = TemplateUploadForm()
+    if form.validate_on_submit():
+        template_name = form.template_name.data
+        file = form.file.data
+        if not os.path.exists(file_upload_path):
+            os.makedirs(file_upload_path)
+        if file:
```
Comment (srh-sloan) [inline app/blueprints/templates/routes.py:40]: Do we need to write the file to disk somewhere? Where will that be on AWS? Can't we just use the in-memory data from the upload and read the json? If we do need to write the file for some reason can we remove it after import please

**Review by srh-sloan — CHANGES_REQUESTED** [review]
Looks good, pulled locally and working, just one query about the file

File: app/blueprints/templates/routes.py (hunk 1 above)
Comment (NarenderRajuB) [reply app/blueprints/templates/routes.py:40]: There are two options we can use
1) Save file in the container and deleter after the import /use flask-caching create a temp directory and then remove it.
2) If we need to refer to the file imported in future, then we may have to find a way to store in AWS.

Happy to change based on suggestions.

File: app/blueprints/fund_builder/routes.py [hunk 2]
```diff
@@ -72,8 +76,15 @@ def section(round_id):
         "round_id": str(round_id),
     }
     existing_section = None
+    # TODO there must be a better way than a pile of ifs...
     if request.args.get("action") == "remove":
-        delete_section(section_id=request.args.get("section_id"), cascade=True)
+        delete_section_from_round(round_id=round_id, section_id=request.args.get("section_id"), cascade=True)
```
Comment (nuwan-samarasinghe) [inline app/blueprints/fund_builder/routes.py:81]: Shall we add some logs statements in to the critical points that will be easy for us to track down bugs also we'll add some comments

File: app/blueprints/templates/routes.py (hunk 1 above)
Comment (nuwan-samarasinghe) [reply app/blueprints/templates/routes.py:40]: As we discussed @NarenderRajuB  go ahead with the change & also check whether temp dir being deleted once done other wise these files will get piled up until a restart

File: tests/test_db.py [hunk 3]
```diff
@@ -288,4 +298,293 @@ def test_form_sorting(seed_dynamic_data, _db):
...
     assert result_section.forms[2].form_id == formX.form_id
     assert result_section.forms[3].form_id == form2.form_id
-    assert result_section.forms[3].section_index == 3
+    assert result_section.forms[3].section_index == 4
+
+
+section_id = uuid4()
+
+
+@pytest.mark.seed_config(
```
Comment (nuwan-samarasinghe) [inline tests/test_db.py:307]: Can we rename the file into `test_db_seed` or similar thing when we look the file name it seems like going to write db level tests

**Review by Adam-W1 — APPROVED** [review]
LGTM

File: app/blueprints/templates/routes.py [hunk 4]
```diff
@@ -12,9 +19,32 @@
     template_folder="templates",
 )
 
+file_upload_path = os.path.join(os.path.dirname(__file__), "uplaoded_files")
```
Comment (nuwan-samarasinghe) [inline app/blueprints/templates/routes.py:22]: DO we still need to upload & create a file inside the container cant we read the maltipartfile it self

**Review by nuwan-samarasinghe — CHANGES_REQUESTED** [review]

File: app/blueprints/templates/routes.py (hunk 4 above)
Comment (NarenderRajuB) [reply app/blueprints/templates/routes.py]: Thanks @nuwan-samarasinghe  removed this, this was added earlier to save file, which is not used now

**Review by nuwan-samarasinghe — APPROVED** [review]
LGTM

**Pull request merged**